client/src/use/useModeManager.ts

```diff
@@ -0,0 +1,137 @@
+import { Ref, computed } from '@vue/composition-api';
+import Track, { TrackId } from '@/lib/track';
+import { RectBounds } from '@/utils';
+import { NewTrackSettings } from './useSettings';
+
+export interface Seeker {
+    seek(frame: number): void;
+    nextFrame(): void;
+  }
+/**
+ * The point of this composition function is to define and manage the transition betwee
+ * different UI states within the program.  States and state transitions can be modified
+ * based on settings, blocked if it tries to go to incompatible state or provide feedback
+ *
+ * Mostly allows us to inject additional logic into transitions.
+ */
+export default function useModeManager({
+  selectedTrackId,
+  editingTrack,
+  frame,
+  trackMap,
+  playbackComponent,
+  newTrackSettings,
+  selectTrack,
+  getTrack,
+  selectNextTrack,
+  addTrack,
+  removeTrack,
+}: {
+    selectedTrackId: Ref<TrackId | null>;
+    editingTrack: Ref<boolean>;
+    frame: Ref<number>;
+    trackMap: Map<TrackId, Track>;
+    playbackComponent: Ref<Seeker>;
+    newTrackSettings: NewTrackSettings;
+    selectTrack: (trackId: TrackId | null, edit: boolean) => void;
+    getTrack: (trackId: TrackId) => Track;
+    selectNextTrack: (delta?: number) => TrackId | null;
+    addTrack: (frame: number, defaultType: string) => Track;
+    removeTrack: (trackId: TrackId) => void;
+}) {
+  function handleSelectTrack(trackId: TrackId | null, edit = false) {
+    selectTrack(trackId, edit);
+  }
+  //Handles adding a new track with the NewTrack Settings
+  function handleAddTrack() {
+    selectTrack(addTrack(frame.value, newTrackSettings.type).trackId, true);
+  }
+
+  function handleTrackTypeChange({ trackId, value }: { trackId: TrackId; value: string }) {
+    getTrack(trackId).setType(value);
+  }
+  // Default settings which are updated by the CreationMode component
+  // Not making them reactive, and eventually will probably be in localStorage
+
+
+  function newTrackSettingsAfterLogic(newTrack: Track) {
+    if (newTrack && newTrackSettings !== null) {
+      if (newTrackSettings.mode === 'Track' && newTrackSettings.modeSettings.Track.autoAdvanceFrame) {
+        playbackComponent.value.nextFrame();
+      } else if (newTrackSettings.mode === 'Detection') {
+        if (newTrackSettings.modeSettings.Detection.continuous) {
+          handleAddTrack();
+        } else { //Deselect the new track
+          selectTrack(newTrack.trackId, false);
+        }
+      }
+    }
+  }
+
+  function handleUpdateRectBounds(frameNum: number, bounds: RectBounds) {
+    if (selectedTrackId.value !== null) {
+      const track = trackMap.get(selectedTrackId.value);
+      if (track) {
+        const features = track.getFeature(frameNum);
+        let newTrack = false;
+        if (!features || features.bounds === undefined) {
+        //We are creating a brand new track and should apply the newTrackSettings
+          newTrack = true;
+        }
```

Comment:
I'm adding in the `newTrackMode` variable which will be triggered everytime the user goes to make a new track, but we still need to check features to see if a new Detection is being made for the specific frame.  The only way around this would be to watch a bunch of other items with a mess of logic to determine if the user is truly adding a new track or if they are editing a current detection.

- `SelectTrack` would have to monitor for changes to the TrackId and editing type, where if the user goes to a new TrackId and remains in editing mode it needs to turn `newTrackMode` off.
- `frame` would need a watcher for the  the case that user clicks on +Track adds a few advance frames then goes back to edit the frame.  We need to know that that specific frame already has a detection associated with it so we don't advance forward directly after the edit.  So the `SelectTrack` logic wouldn't work because it is the same track and the same mode, just a different frame.  But if the user goes forward in time a bunch of frames, it is a frame without a detection and we would want to advance the frame, or remain in continuous detection mode.